windows/src/desktop/kmshell/main/Keyman.System.UpdateStateMachine.pas

```diff
@@ -753,18 +754,33 @@ procedure DownloadingState.Enter;
 var
   DownloadResult: Boolean;
   RetryCount: Integer;
+  FMutex: TKeymanMutex;
 begin
   // Enter DownloadingState
   bucStateContext.SetRegistryState(usDownloading);
 
   RetryCount := 0;
   DownloadResult := False;
```

Comment:
I think these should be immediately before their first use in the repeat/until loop, and then DownloadResult does not need to be pre-initialized either

Reply:
```suggestion
```